### PR TITLE
ROX-19860: Reduce RBAC resource requirements for routes

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

TL;DR reduce role-based access control resource requirements for 6 route keys:
* `compliance`
* `configmanagement`
* `'network-graph'` indirectly via `nonGlobalResourceNamesForNetworkGraph`
* `'policy-management'`
* `risk`
* `'vulnerability-management'`

For cherry pick into 4.2.1 release branch, changes are limited to prevent merge conflicts with 3 route keys added in routePaths.ts file **after** 2023-08-24 https://github.com/stackrox/stackrox/pull/7566
* `'administration-events'`
* `'clusters/init-bundles'`
* `'deferral-configuraton'`

### Problem

Because I missed the forest for the trees while adding conditional rendering within routes to reduce `resourceAccessRequirements` during sprint 4.2C, some routes became unavailable for primary use cases because of lack of resources for secondary use cases, especially for Vulnerability Management 1.0 route.

#### Vulnerability Management 1.0

1. WorkflowAdministration
    * VulnMgmt/Entity/Cluster does not use `policyStatus` nor `policyCount` values.
    * VulnMgmt/Entity/Namespace does not use `policyStatus` nor `policyCount` values.
2. Alert
    * Clusters, Namespaces, and Deployments use `policyStatus` and `latestViolation` values.
3. Cluster
    * ClustersWithNostClusterVulnerabilities renders database error.
    * CvesMenu renders `undefined` for all values.
    * /main/vulnerability-management/clusters route renders PageNotFound element.
4. Namespace
    * TopRiskiestEntities renders database error.
    * /main/vulnerability-management/namespaces route renders PageNotFound element.
5. Node
    * TopRiskiestEntities renders database error.
    * CvesMenu renders `undefined` for all values.
    * /main/vulnerability-management/nodes route renders PageNotFound element.

#### Compliance 1.0

With Compliance only:
* In addition to READ_WRITE_ACCESS for Compliance, **Scan environment** requires READ_ACCESS for at least one of Cluster, Namespace, Node.
* However, if environment has been scanned, absence of those resources is affects only specific widgets or pages.

#### Configuration Management

1. Unlike most other routes, it is not clear which resource (or subset of resources) are primary.
2. Dashboard is a mess even with all 4 resources directly related to its widgets.

#### Policy Management

Although, policies list and policy detail page make GET /v1/notifiers request, consider it secondary.

#### Network Graph

1. DeploymentExtension is required for 2 tabs of the side panel and IN THE FUTURE verify nowhere else:
    * Flows: GET /v1/networkbaseline/id has 403 Forbidden status
    * Baselines: ditto
2. NetworkPolicy: IN THE FUTURE investigate whether it is required only for **Network Policy** tab of side panel.

#### Listening Endpoints

With Deployment but not DeploymentExtension resource: page renders infinite polling loop.

#### Risk

With Deployment but not DeploymentExtension resource:
* Can render list page.
* Cannot render side panel, because it needs additional conditional rendering.
    * `fetchProcesses` request requires DeploymentExtension resource:
        https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Risk/RiskSidePanel.js#L27
    * `ProcessDetails` tab requires DeploymentExtension in addition to Alert resource.
        https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Risk/RiskSidePanelContent.js#L77-L98

#### Vulnerability Management 1.0

1. Delete WorkflowAdministration resource now and then IN THE FUTURE:
    * VulnMgmt/Entity/Cluster/VulnMgmtEntityCluster.js delete unused properties from query.
    * VulnMgmt/Entity/Cluster/VulnMgmtEntityNamespace.js delete unused properties from query.
2. Comment out resources to leave the same minimum resources as Workload CVEs route.

#### Compliance 1.0

1. Comment out resources to leave same minimum resources as Compliance 2.0 route.

#### Configuration Management

1. Comment out resources to leave minimum requirement of at least one that is directly related to a dashboard widget.

#### Policy Management

1. Comment out resources except for WorkflowAdministration, even though clone/create/edit workflow will probably not work correctly without Deployment, Image, Integration.

#### Network Graph

1. Comment out DeploymentExtension and NetworkPolicy resources and IN THE FUTURE add conditional rendering.
2. Leave Deployment because required for **Details** tab of side panel.

#### Risk

1. Delete DeploymentExtension resource and IN THE FUTURE make request and tab conditional.

### Residue

IN THE FUTURE, investigate pro and con of conditional code:
1. CvesMenu on Vulnerability Management 1.0 dashboard for which the request is all or nothing for 3 resources.
2. Policies
    * For policies list and policy detail page.
    * Evaluate pro and con for clone/create/edit workflow to require the additional resources: Deployment, Image, Integration.
3. SummaryCounts on main dashboard for which the request was all or nothing for 6 resouces.
4. Investigate whether global search can conditionally render partial results.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~
- [x] ~Evaluated and added CHANGELOG entry if required~ Release Notes instead of CHANGELOG

### Manual testing

Adjust user role permissions from another browser window.

#### Vulnerability Management 1.0

1. Without Alert resource: see **Cannot find the whatever** error page.

    * Visit /main/vulnerability-management/clusters
    * Visit /main/vulnerability-management/namespaces
    * Visit /main/vulnerability-management/deployments

2. Without Cluster resource:

    * See `undefined` for all 3 counts in CvesMenu.
    * See database error in **Clusters and so on** widget.
        clusterVulnerabilityCount

3. Without Namespace resource:

    * See database error in **Top risky namespaces** widget.
        TopRiskiesEntities_Namespace

4. Without Node resource:

    * See `undefined` for all 3 counts in CvesMenu.
    * See database error in **Top risky nodes** widget.
        nodeVulnerabilityCount

5. Without WorkflowAdministration resource:

    * Visit /main/vulnerability-management/clusters and then click a row: see error before deletion and page after.
        However, do not include this change.
        VulnMgmtEntityCluster

    * Visit /main/vulnerability-management/namespaces and then click a row: see error before deletion and page after.
        However, do not include this change.
        VulnMgmtEntityNamespace

#### Compliance 1.0

With Compliance resource only:

1. With READ_WRITE_ACCESS, click **Scan environment** button.

    * See triggerScan request with `complianceTriggerRuns: []` in response.
        Compliance_only_without_scan

2. With READ_ACCESS after a scan.

    * See data for only some widgets.
        Compliance_only_with_scan

3. Add Cluster resource

    * See data for additional widget.
        Compliance_Cluster_1

    * See data for additional widgets. My notes failed me how these look without Cluster resource :(
        Compliance_Cluster_2

3. Visit /main/compliance/clusters

    * See list, but errors for side panel.
        Compliance_Cluster_cluster

5. Visit /main/compliance/namespaces

    * See absence of data for list.
        Compliance_Cluster_namespaces

#### Configuration Management

1. Visit /main/configuration-management

    * See a mess of errors, even with all 4 of the `someResource` resources.
        Configuration_Management_someResource

#### Policy Management

1. Visit /main/policy-management/policies

    * See GET /v1/notifiers request which has 403 Forbidden status code.
        WorkflowAdmnistration_Policies

2. Click a policy

    * See GET /v1/notifiers request which has 403 Forbidden status code.
        WorkflowAdministration_Policy

#### Network Graph

1. Visit /main/network-graph,click a deployment, and then click **Flows** tab. Ditto for **Baselines** tab.

    * See absence of data without DeploymentExtension resource.
        DeploymentExtension_Network_Flows

#### Listening Endpoints

1. Visit /main/listening-endpoints without DeploymentExtension resource.

    * See primary requests have 403 Forbidden status code.
        DeploymentExtension_Listening_Endpoints

#### Risk

1. Visit /main/risk without DeploymentExtension resource.

    * See list but no tabs in side panel.
        DeploymentExtention_Risk

### CODE REVIEW

**Review by vjwilson — APPROVED** [review]
I'm fine with this implementation, if @connorgorman agrees this will cover his concerns for now.

Two minor questions:
1. The PR is into master, but we had talked about PR-ing into the 4.2.x release branch. Is this to avoid bifurcating the code paths, since we plan to improve the implementation before 4.3 is cut?
2. Do we need Watched Images as one of the required fields in VM ?

**Comment by pedrottimark** [comment]
1. Branch from master for clarity, simplicity, and ongoing testability, because changes are independent of changes to routePaths.ts file so far during 4.3 sprints.
2. Good question why 3 resources for Vulnerability Management 1.0
    After we get CI run tonight, I will repeat steps in the morning and add screenshots to Testing Performed
    * WatchedImage for images and/or image-cves list
    * Deployment for image and/or image-cve side panel
    * Bonus I will also confirm behavior of corresponding single pages.

File: ui/apps/platform/src/routePaths.ts [hunk 1]
```diff
@@ -291,15 +294,15 @@ const routeRequirementsMap: Record<RouteKey, RouteRequirements> = {
...
-            'Cluster',
+            // 'Alert', // for Cluster and Deployment and Namespace
+            // 'Cluster',
             'Deployment',
             'Image',
-            'Namespace',
-            'Node',
+            // 'Namespace',
+            // 'Node',
             'WatchedImage', // for Image
```
Comment (connorgorman) [inline ui/apps/platform/src/routePaths.ts:305]: Although it's required, it really should not be in order to view images so I would recommend that we remove it as the errors are pretty clear that you need watchedimage rights. We will file a ticket to clean up this permission requirement

**Comment by connorgorman** [comment]
One random thing I noticed was that the deployment tab was removed from the violations page once you click into a deployment. Do you where that logic was implemented? I don't know when we changed that and the data is returned in the API but the tab doesn't show up in the UI if you don't have Deployment resource access

**Comment by pedrottimark** [comment]
> One random thing I noticed was that the deployment tab was removed from the violations page once you click into a deployment. Do you where that logic was implemented? I don't know when we changed that and the data is returned in the API but the tab doesn't show up in the UI if you don't have Deployment resource access

Good eyes, Connor. https://github.com/stackrox/stackrox/pull/7411

Although alerts response indeed has a deployment, here is a different request that requires Deployment resource:

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Violations/Details/Deployment/DeploymentDetails.tsx#L26-L28

The reason to omit the tab is React does not allow `useFetchDeployment(alertDeployment.id)` in an `if` statement.

The way that I reread the code, only the **Network policies** card could have been rendered in that situation:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Violations/Details/Deployment/DeploymentDetails.tsx#L84-L91

And that card is more about **namespaces** than the **deployment** itself:
Network_policies

Therefore, maybe better as a separate conditionally rendered tab if NetworkPolicy resource and if any policies?
Violation_Deployment

File: ui/apps/platform/src/routePaths.ts (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/routePaths.ts]: Here are relevant screenshots that I had forgotten to take yesterday.

1. Visit /main/vulnerability-management with only Image and not Deployment resource.

    Keep Deployment resource as required because image CVEs are closely related to deployments, therefore data errors for widgets:
    * Top risky deployments by CVE count and CVSS score
    * Recently detected image vulnerabilities
    * Most common image vulnerabilities
        Deployment_Dashboard

2. Visit /main/vulnerability-management/images with Image and not WatchedImage resource.

    * Although data error prevents rendering the list, I commented out the resource as requested above.
        WatchedImage_Images

**Comment by pedrottimark** [comment]
ocp-4-10-ui-e2e-tests

> kubelet may be retrying requests that are timing out in CRI-O due to system load. Currently at stage sandbox storage creation: context deadline exceeded
